Update 2020-10-10.interspeciesio-ml-workshop.md (#36)
earthspecies/project: events/2020-10-10.interspeciesio-ml-workshop.md

# October 10, 2020 - Earth Species Project Machine Learning & Bioacoustics Workshop

> ## hosted by Interspecies.IO

Reviewer: I wonder how this will render. My fear is that it will be a bit kludgy. I went back and forth on this. Let's find out!

> It was a tremendous honor to present to the [Interspecies.io](https://interspecies.io/) community and share our latest work. The workshop was hosted by Interspecies.io, a multidisciplinary group founded by Peter Gabriel (musician and activist), Dr. Vint Cerf (Vice President at Google and Co-designer of the Internet), Dr. Diana Reiss (Professor at Hunter College, a cognitive psychologist, and marine mammal scientist), and Dr. Neil Gershenfeld (head of MIT’s Center for Bits and Atoms).

Reviewer: ```suggestion
It was a tremendous honor to present to the [Interspecies.io](https://interspecies.io/) community and share our latest work. The workshop was hosted by Interspecies.IO, a multidisciplinary group founded by Peter Gabriel (musician and activist), Dr. Vint Cerf (Vice President at Google and Co-designer of the Internet), Dr. Diana Reiss (Professor at Hunter College, a cognitive psychologist, and marine mammal scientist), and Dr. Neil Gershenfeld (head of MIT’s Center for Bits and Atoms).
```

> ## Attendees: 

Reviewer: ```suggestion
## Attendees
```

> Nearly 100 biologists, computational linguists, information theorists, musicians, and technologists. Continue the conversation with us [here](https://github.com/earthspecies/project/discussions/35). 

Reviewer: - Do we even need this? Feels like tooting our own horn a bit. Not super opinionated so I defer to you + @aza.
- If we keep, I'm not a huge fan of the term technologists. My suggestion included in this comment.

```suggestion
Nearly 100 biologists, computational linguists, computer scientists and engineers, information theorists, musicians, and artists. Continue the conversation with us [here](https://github.com/earthspecies/project/discussions/35). 
```

Author: I didn't know how we would best refer to folks like Vint C., Neil and ML researchers on the call. Technologists was the best catch-all I could come up with but I prefer your language.

Reviewer 2: I do think it is nice to call out the audience. It's more of a tooting Interspecies.io horn in a way ;)

> ## Recording: 

Reviewer: ```suggestion
## Recording
```

[![video thumb](https://i.imgur.com/XwZq1TS.png)](https://archive.org/details/20201010-interspecies-i-o-earth-species-project-ai-toolbox)

> ## Opensource Code: 

Reviewer: Open source is either: ```open source``` or ```open-source```, but not ```opensource```.

```suggestion
## Resources Covered in Talk 
All of our work is open-source and freely available. If you have any questions or want help using this in your research, please open an Issue in the respective repository.
```

Author: Love this edit!

 
 * [Cocktail Party Problem](https://github.com/earthspecies/cocktail-party-problem)
 * [Acoustic representation toolbox ](https://github.com/earthspecies/acoustic-representation-toolbox)
 * [Denoising, detection, and classification](https://github.com/earthspecies/dolphin-strandings)
 * [ESP Library](https://github.com/earthspecies/library)

## Participate: 

> The Earth Species Library is a collection of universally accessible bioacoustics datasets, ranging from humpback whales to zebra finches, tuned for machine learning. These datasets will enable machine learning researchers to train and validate novel machine learning methodologies and develop increasingly complex approaches to answer some of biology's most elusive questions. If you are interested in contributing your bioacoustic data and annotations to the Earth Species Library, please get in touch with [katie@earthspecies.org](mailto:katie@earthspecies.org). 

Reviewer: ```suggestion
The [Earth Species Library](https://github.com/earthspecies/library) is a collection of universally accessible bioacoustics datasets, ranging from humpback whales to zebra finches, tuned for machine learning. These datasets will enable machine learning researchers to train and validate novel machine learning methodologies and develop increasingly complex approaches to answer some of biology's most elusive questions. If you are interested in contributing your bioacoustic data and annotations to the Earth Species Library, please get in touch with [katie@earthspecies.org](mailto:katie@earthspecies.org). 
```

Author: Awesome edit!

## Getting Started: 

*   [Introduction to using Collab](https://colab.research.google.com/notebooks/intro.ipynb)
*   [Digital Signal Processing with Python](https://github.com/earthspecies/intro-to-DSP-with-python)

> ## Resources: 

Reviewer: I would roll getting started into resources.

Reviewer 3: Yup! Looks good - maybe just 'Introduction to Colab' vs Collab

> ## Getting Started: 
> 
> *   [Introduction to using Collab](https://colab.research.google.com/notebooks/intro.ipynb)
> *   [Digital Signal Processing with Python](https://github.com/earthspecies/intro-to-DSP-with-python)
> 
> ## Resources: 

Reviewer: ```suggestion
## Getting Started: 

*   [Introduction to using Collab](https://colab.research.google.com/notebooks/intro.ipynb)
*   [Digital Signal Processing with Python](https://github.com/earthspecies/intro-to-DSP-with-python)

## Resources
```

*   [Music Source Separation in the Waveform Domain](https://arxiv.org/abs/1911.13254)
*   [Deep Audio Prior](https://arxiv.org/abs/1912.10292)
*   [Deep Learning for Monaural Speech Separation](https://paris.cs.illinois.edu/pubs/huang-icassp2014.pdf)
*   [The Separation of Overlapped Dolphin Signature Whistle Based on Blind Source Separation](https://ieeexplore.ieee.org/document/8242534)
*   [A Comparative Study of Blind Source Separation for Bioacoustics Sounds based on FastICA, PCA and NMF](https://www.sciencedirect.com/science/article/pii/S1877050918312468?via%3Dihub)
*   [Investigating Deep Neural Transformations for Spectrogram-Based Musical Source Separation](https://arxiv.org/abs/1912.02591)
*   [A Physiologically Inspired Model for Solving the Cocktail Party Problem](https://link.springer.com/content/pdf/10.1007/s10162-019-00732-4.pdf)
*   [Audio Source Separation with Discriminative Scattering Networks](https://arxiv.org/abs/1412.7022)
*   [Blind Nonnegative Source Separation Using Biological Neural Networks](https://arxiv.org/abs/1706.00382)

# 

> **Miscellaneous Key Resources**

Reviewer: What's the difference between these and the things in getting started? Can we combine?

Reviewer 3: We could probably remove the Misc. Key Resources title - the resources above were kind of ordered in terms of relevance, and then as I proceeded through the literature, I randomly added more to the Misc. Key Resources section to focus on a smaller fraction of the 400-500 total

Author: Sounds good @pcbermant , I'll remove the title.

*   [A Comprehensive Study of Speech Separation: Spectrogram vs Waveform Separation](https://arxiv.org/abs/1905.07497)
*   [FurcaNeXt: End-to-End Monaural Speech Separation with Dynamic Gated Dilated Temporal Convolutional Networks](https://arxiv.org/abs/1902.04891)
*   [Improving Universal Sound Separation Using Sound Classification](https://arxiv.org/abs/1911.07951)
*   [SpEx: Multi-Scale Time Domain Speaker Extraction Network](https://arxiv.org/abs/2004.08326)
*   [Meta-Learning Extractors for Music Source Separation](https://arxiv.org/abs/2002.07016)
*   [Unsupervised Learning of Semantic Audio Representations](https://arxiv.org/abs/1711.02209)
*   [Sudo rm -rf: Efficient Networks for Universal Audio Source Separation](https://arxiv.org/abs/2007.06833)
*   [Unsupervised Sound Separation Using Mixtures of Mixtures](https://arxiv.org/abs/2006.12701)
*   [Listen to What You Want: Neural Network-based Universal Sound Selector](https://arxiv.org/abs/2006.05712)
*   [Speech Separation Based on Multi-Stage Elaborated Dual-Path Deep BiLSTM with Auxiliary Identity Loss](https://arxiv.org/abs/2008.03149)

> *   [Identify Speakers in Cocktail Parties with End-to-End Attention](https://arxiv.org/abs/2005.11408)

Reviewer: This is frickin epic. @pcbermant, anything to add here?

Author: Those are all directly from @pcbermant 's readme in the Cocktail Party bookshelf, so I would assume he's good with it for us to push live?

Reviewer 3: Is it worth linking directly to the bookshelf? There are a ton of other resources in there, but these ones seemed to be some of the most noteworthy.

Reviewer 2: This is so good!

# 

> **Machine Learning Concepts**

Reviewer: Per the other review and @pcbermant's feedback
```suggestion
**Experimental Audio Source Separation Concepts to Explore**
```

*   Matched Filters
*   Hilbert Huang Transforms
*   Lombard Effect
*   Dip-Listening Hypothesis
*   Noise-Invariant, Rule-Encoding, and Temporal Edge Neurons
*   FastICA (Independent Component Analysis), Point Source Separation (PSS), Non-Negative Matrix Factorization (NMF)
*   Cortical Representation of Speech, Auditory Object Analysis
*   Deep Scattering Spectrum, Kymatio, Invariant Scattering Convolution Networks
*   Error-Gated Hebbian Rule
*   Probabilistic Latent Component Analysis (PLCA), Robust PCA (RPCA), Low-Rank Modeling
*   Progressive Learning
*   Denoising Autoencoders
*   Temporal Convolutional Networks (TCNs)
*   Perfect Reconstruction Filterbank (PRFB), Gammatone Filterbank
*   Deep Feature Losses, Short-Time Objective Intelligibility (STOI), Perceptual Evaluation of Speech Quality (PESQ)
*   Wiener Filtering
*   Permutation Invariant Training
*   Supervised Independent Vector Analysis
*   Phase Spectra and Group Delay
*   Spectrogram Fusion
*   Self-Organizing Background Subtraction
*   Phase-Aware Signal Processing
*   Deep Clustering, Deep Attractor, and Selective Hearing Networks

> *   Multimodal/Crossmodal Fusion

Reviewer: These are not all deep learning concepts ... are these things that @pcbermant mentioned that we're just including as a reference?

> *   Multimodal/Crossmodal Fusion

Reviewer: These are not all deep learning concepts ... are these things that @pcbermant mentioned that we're just including as a reference?

@pcbermant, thoughts?

Reviewer 3: Yeah - these are super rough notes for ideas worth exploring during the CPP investigation. What about a title like ~ "Experimental Audio Source Separation Concepts to Explore"

Reviewer: I like.

Author: I like that too!
